proposals/NNNN-add-generic-hmi-plugin-support.md

```diff
@@ -0,0 +1,362 @@
+        SidebarEnabled: true, //Plugin application has a UI component, and should be displayed beside the main HMI
+        SidebarName: 'Voice', //Title to display alongside the plugin UI
+        MenuEnabled: true, //Plugin has a custom screen within the Generic HMI, accessible via menu
+        MenuName: 'Voice Commands' //Title to display in the menu for this plugin's screen
+    },
+    ...
+};
+```
+
+**Note:** This format is tentative and could change depending on the final implementation
+
+### Example Plugins
+
+Several example plugins should be developed as a baseline for testing purposes, these can also be used as a reference for production implementations. This section details the functionality required in each plugin to allow for full feature coverage in the Generic HMI.
+
+#### Voice Recognition Plugin
+
+Connects to the `VR` component of SDL Core, overriding this component in the Generic HMI.
+
+*Features:*
+
+- General interface functions
+    - VR.IsReady
+    - VR.GetCapabilities
+- Voice commands (both "Command" and "Choice" type), this feature can be ported from Manticore
+    - VR.AddCommand
+    - VR.DeleteCommand
+    - VR.PerformInteraction
+    - VR.OnCommand
+- Voice recognition state control
+    - VR.Started
+    - VR.Stopped
+- Language switching support
+    - VR.OnLanguageChange
+    - VR.GetLanguage
+    - VR.GetSupportedLanguages
+    - VR.ChangeRegistration
+- The "Help" command, which will trigger the TTS component to speak the app's `helpPrompt`
+
+#### Vehicle Info Plugin
+
+Connects to the `VehicleInfo` component of SDL Core, overriding this component in the Generic HMI.
+
+*Features:*
+
+- General interface functions
+    - VehicleInfo.IsReady
+- Vehicle data, this feature can be ported from Manticore
+    - VehicleInfo.GetVehicleData
+    - VehicleInfo.SubscribeVehicleData
+    - VehicleInfo.UnsubscribeVehicleData
+    - VehicleInfo.OnVehicleData
+    - VehicleInfo.GetVehicleType
+- Raw vehicle info requests
+    - VehicleInfo.ReadDID
+    - VehicleInfo.GetDTCs
+    - VehicleInfo.DiagnosticMessage
+
+#### Remote Control Plugin
+
+Connects to the `RC` component of SDL Core, overriding this component in the Generic HMI. This could potentially be split into several plugins (by module type) depending on the system.
+
+*Features:*
+
+- General interface functions
+    - RC.IsReady
+    - RC.GetCapabilities
+- Remote Control data, this feature can be ported from Manticore
+    - RC.SetInteriorVehicleData
+    - RC.GetInteriorVehicleData
+    - RC.OnInteriorVehicleData
+- Remote Control settings and status control
+    - RC.OnRCStatus
+    - RC.OnRemoteControlSettings
+    - RC.SetGlobalProperties
+    - RC.GetInteriorVehicleDataConsent
+- Alternate audio source (ex. radio)
+
+#### Buttons Plugin
+
+Connects to the `Buttons` component of SDL Core. This plugin would also communicate ButtonPress events to other plugins where needed.
```

Comment:
>ButtonPress events

Should "ButtonPress" be two words here, or does it need backticks around it?

Reply:
ButtonPress is a command, so backticks would be appropriate

Reply:
Fixed in https://github.com/smartdevicelink/sdl_evolution/pull/1167/commits/9b586573519b27d6d0de1498d4b7f44664223dc4